### ISSUE TITLE
how to fine-tune the original external parameters?

### ISSUE THREAD

**Issue by fanDongHust** [SokratG/Surround-View#6]
When the lens is corrected, on the actual road, the orientation of the lens changes, because the road is uneven, or the tire pressure is unstable, etc., the external parameters calibrated by the chessboard pattern is not valid no longer, then how to fine-tune the original external parameters, to reduce the error of the splicing results?
![image](https://user-images.githubusercontent.com/20364891/166080915-9f1a3dec-4494-421e-ba57-720b48603c42.png)

Thank you!

**Comment by SokratG** [comment]
> 

This is not trivial or not well defined task. You compute some geometrical parameters of your cameras and then change them, ofc, you must re-compute new parameters. I tried found the solution of this task, when I research this area and didn't found unique solution. My suggestion how it could be solved:
1. Use online(auto-) calibration technique by auto detecting feature points and compute camera parameters by points. But this can be computationally intensive and dependent on the computational resources of your hardware platform.
2. Also I found this [presentation](https://on-demand.gputechconf.com/gtc/2015/presentation/S5295-Miguel-Sainz-Timo-Stich.pdf) - they use visual odometry for reduce image displacement, but I not well understand the idea and don't found the papers about this. If you understand that's may help.
3. Use Deep Learning for warp your camera image for get a good view - this is may be hard and nobody did it. It would be a novel and it has a interesting research area.

**Comment by SokratG** [comment]
> When the lens is corrected, on the actual road, the orientation of the lens changes, because the road is uneven, or the tire pressure is unstable, etc., the external parameters calibrated by the chessboard pattern is not valid no longer, then how to fine-tune the original external parameters, to reduce the error of the splicing results? ![image](https://user-images.githubusercontent.com/20364891/166080915-9f1a3dec-4494-421e-ba57-720b48603c42.png)
> 
> Thank you!

This is not trivial or not well defined task. You compute some geometrical parameters of your cameras and then change them, ofc, you must re-compute new parameters. I tried found the solution of this task, when I research this area and didn't found unique solution. My suggestion how it could be solved:
1. Use online(auto-) calibration technique by auto detecting feature points and compute camera parameters by this points. But this can be computationally intensive and dependent on the computational resources of your hardware platform.
2. Also I found this [presentation](https://on-demand.gputechconf.com/gtc/2015/presentation/S5295-Miguel-Sainz-Timo-Stich.pdf) - they use visual odometry for reduce image displacement, but I not well understand the idea and not found the papers about this. If you understand that's may help.
3. Use Deep Learning for warp your camera image for get a good view - this is may be hard and nobody did it. It would be a novel and it has a interesting research area.
